From cbc4887979bd518f672606459756a3889d72b3cb Mon Sep 17 00:00:00 2001
From: Your Name <your.email@example.com>
Date: Sat, 17 May 2025 02:03:03 +0000
Subject: [PATCH] Prevents stub creation for files ending with "_ExecutionStub.exe" and avoids overwriting existing stubs

---
 src/Update/Program.cs | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/src/Update/Program.cs b/src/Update/Program.cs
index 51f5e2cb..cbc48879 100644
--- a/src/Update/Program.cs
+++ b/src/Update/Program.cs
@@ -632,12 +632,20 @@ bool isPEFileSigned(string path)
         async Task createExecutableStubForExe(string fullName)
         {
-            var exe = Utility.FindHelperExecutable(@"StubExecutable.exe");
+            if(fullName.EndsWith("_ExecutionStub.exe")) {
+                return;
+            }
 
             var target = Path.Combine(
                 Path.GetDirectoryName(fullName),
                 Path.GetFileNameWithoutExtension(fullName) + "_ExecutionStub.exe");
 
+            // Return early if the target already exists
+            if (File.Exists(target)) {
+                return;
+            }
+
+            var exe = Utility.FindHelperExecutable(@"StubExecutable.exe");
             await Utility.CopyToAsync(exe, target);
 
             await Utility.InvokeProcessAsync(